engine/src/test/java/org/destinationsol/game/ShardTest.java

```diff
@@ -0,0 +1,144 @@
+        // Default position of dummy bodies is (1f, 0f)
+        // Epsilon testing because of float accuracy
+        assertTrue(constantShard.getPosition().epsilonEquals(1f, 0f, 0.01f));
+    }
+
+    @Test
+    public void toFarObject() {
+        // For performance reasons, shards should not persist while offscreen.
+        // They are pretty tiny in terms of gameplay, after all, so they don't even need to persist
+        assertNull(constantShard.toFarObject());
+    }
+
+    @Test
+    public void getDrawables() {
+        assertEquals(constantShard.getDrawables(), drawables);
+    }
+
+    @Test
+    public void getAngle() {
+        // 30° is the default angle of dummyBody
+        assertEquals(constantShard.getAngle(), 30f, 0.25f);
+    }
+
+    @Test
+    public void getSpeed() {
+        Body body = BodyUtilities.createDummyBody();
+        body.setLinearVelocity(1f, 2f);
+        final Shard shard = new Shard(body, drawables);
+        assertTrue(shard.getSpeed().epsilonEquals(1f, 2f, 0.01f));
+        assertTrue(constantShard.getSpeed().epsilonEquals(0f, 0f, 0.01f));
+    }
+
+    @Test
+    public void handleContact() {
+        // Shards are not big enough to cause damage or get damage or anything, so this just shouldn't crash
+        constantShard.handleContact(new Shard(BodyUtilities.createDummyBody(), drawables), 10f, InitializationUtilities.game, new Vector2(0f, 0f));
+    }
+
+    @Test
+    public void isMetal() {
+        // Shards are so tiny that no sound should be played when one hits anything
+        assertNull(constantShard.isMetal());
+    }
+
+    @Test
+    public void hasBody() {
+        assertTrue(constantShard.hasBody());
+    }
+
+    @Test
+    public void update() {
+        // Just should not throw exception - all the moving and stuff is handled by body, this has just to exist
+        constantShard.update(InitializationUtilities.game);
+    }
+
+    @Test
+    public void shouldBeRemoved() {
+        // This should persist for as long as seen/loaded
+        assertFalse(constantShard.shouldBeRemoved(InitializationUtilities.game));
+    }
+
+    @Test
+    public void onRemove() {
+        // TODO onRemove() should free its resources. How to test that?
+        // I guess this just should not crash
+        new Shard(BodyUtilities.createDummyBody(), drawables).onRemove(InitializationUtilities.game);
+    }
+
+    @Test
+    public void receiveDmg() {
+        // Shards have no health, and thus receiveDamage() just should not crash
+        constantShard.receiveDmg(100, InitializationUtilities.game, null, DmgType.BULLET);
+    }
+
+    @Test
+    public void receivesGravity() {
+        // When ship is shattered in gravity, its shards should fall to the ground
+        assertTrue(constantShard.receivesGravity());
+    }
+
+    @Test
+    public void receiveForce() {
+        // TODO I don't quite know what does this even do, so better leave this for sb else
+        fail("Implement this test!");
```

Comment:
Probably just leave a TODO with the name then, don't implement a failing tests - the tests will fail! :D

Alternatively you can mark the test as ignored while leaving the code in place, then it will just skip till somebody addresses the TODO